utils/fixed_decimal/src/decimal.rs

```diff
@@ -552,6 +552,299 @@ impl FixedDecimal {
         self.check_invariants();
     }
 
+    /// Increments the digits by 1. if the digits are empty, it will add
+    /// an element with value 1.
+    fn increment_abs_by_one(&mut self) -> Result<(), Error> {
+        for digit in self.digits.iter_mut().rev() {
+            *digit += 1;
+            if *digit < 10 {
+                #[cfg(debug_assertions)]
+                self.check_invariants();
+                return Ok(());
+            }
+
+            *digit = 0;
+        }
+
+        if self.magnitude == i16::MAX {
+            self.magnitude = 0;
+            self.digits.clear();
+
+            #[cfg(debug_assertions)]
+            self.check_invariants();
+            return Err(Error::Limit);
+        }
+
+        self.digits.clear();
+
+        // Still a carry, carry one to the next magnitude.
+        self.digits.push(1);
+        self.magnitude += 1;
+
+        if self.upper_magnitude < self.magnitude {
+            self.upper_magnitude = self.magnitude;
+        }
+
+        self.remove_trailing_zeros_from_digits_list();
```

Comment:
Nit: You should not need to call this function here (if you address my other comment). Please remove it.

Reply:
actually, it was wrong because we had already cleared the digits.